agent/agent.go

```diff
@@ -301,6 +302,15 @@ type Agent struct {
 	// Shared RPC Router
 	router *router.Router
 
+	// uiConfig is an atomically held copy of the current UI config. It's
+	// populated from config.UIConfig on agent startup but then becomes the
+	// canonical source where all reads during runtime should be made. It may be
+	// updated on a configuration reload.
```

Comment:
It's populated in `Agent.New`, not `Agent.Start`, right? agent startup feels misleading.

*Why* is this field used instead of `a.config` ? I think that is important to capture in the comment.

Reply:
Yeah I meant "startup" generally - the initial creation of an agent - rather than literally the Start method but I can change the wording if that's confusing.

> Why is this field used instead of a.config ? 

That would be because:

> It may be updated on a configuration reload.

I can make that more explicit though if you didn't read that as a clear enough explanation!